public/app/plugins/datasource/cloudwatch/types.ts

```diff
@@ -10,3 +10,23 @@ export interface CloudWatchQuery extends DataQuery {
   period: string;
   expression: string;
 }
+
+export interface CloudWatchOptions extends DataSourceJsonData {
+  name: string;
+  jsonData: {
+    authType: string;
+    defaultRegion: string;
+    timeField: string;
+    assumeRoleArn: string;
+    database: string;
+    customMetricsNamespaces: string;
+  };
```

Comment:
something seems fishy here -- `DataSourceJsonData` should be the element defined as `jsonData`